File: Lecture4Exercise/allchr_pca.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from matplotlib.backends.backend_pdf import PdfPages

# 1. Load PCA results
pca = pd.read_csv("allchr_unrelated_pruned_pca.eigenvec", sep='\t')
pca.columns = [c.replace('#', '') for c in pca.columns]

# 2. Load Population Metadata
panel = pd.read_csv("integrated_call_samples_v3.20130502.ALL.panel", sep='\t')

# 3. Merge data
df = pd.merge(pca, panel, left_on='IID', right_on='sample')

# 4. Create PDF Report
with PdfPages('allchr_pca.pdf') as pdf:
    plt.figure(figsize=(12, 8))
    
    # Plotting by 'pop' (Specific populations like YRI, CEU, etc.)
    # Change 'hue' to 'super_pop' for Continental clusters
    sns.scatterplot(data=df, x='PC1', y='PC2', hue='pop', 
                    palette='turbo', s=40, alpha=0.8, edgecolor='none')
    
    plt.title("1000 Genomes PCA (PC1 vs PC2)")
    plt.xlabel("PC1")
    plt.ylabel("PC2")
    
    # Place legend outside the plot for clarity
    plt.legend(bbox_to_anchor=(1.05, 1), loc='upper left', ncol=2, fontsize='small')
    plt.grid(True, linestyle='--', alpha=0.3)
    
    plt.tight_layout()
    pdf.savefig()
    plt.close()

print("PCA plots saved to PDF.")

File: Lecture4Exercise/allchr_pca.R
library(data.table)
library(ggplot2)

# 1. Load PCA results (PLINK 2.0 format)
pca <- fread("allchr_unrelated_pruned_pca.eigenvec")
# Remove the '#' from the first column name
setnames(pca, "#FID", "FID")

# 2. Load Population Metadata
panel <- fread("integrated_call_samples_v3.20130502.ALL.panel")

# 3. Merge data on Sample ID
# PCA 'IID' matches Panel 'sample'
df <- merge(pca, panel, by.x = "IID", by.y = "sample")

# 4. Generate PDF
pdf("allchr_pca.pdf", width = 10, height = 8)

ggplot(df, aes(x = PC1, y = PC2, color = pop)) +
  geom_point(alpha = 0.7, size = 1.5) +
  theme_minimal() +
  labs(title = "1000 Genomes PCA: PC1 vs PC2",
       subtitle = "Colored by Population (Sub-population)",
       x = "Principal Component 1",
       y = "Principal Component 2",
       color = "Population") +
  # Optional: use super_pop for broader clustering (AFR, EUR, etc.)
  # aes(color = super_pop) 
  guides(color = guide_legend(ncol = 2)) # Make legend readable

dev.off()
